ipywidgets/widgets/widget_templates.py

```diff
@@ -0,0 +1,300 @@
+    center = Instance(Widget, allow_none=True)
+
+    def __init__(self, **kwargs):
+        super(AppLayout, self).__init__(**kwargs)
+        self._update_layout()
+
+    def _update_layout(self):
+
+        grid_template_areas = [["header", "header", "header"],
+                               ["left-sidebar", "center", "right-sidebar"],
+                               ["footer", "footer", "footer"]]
+        grid_template_columns = ['1fr', '2fr', '1fr']
+        grid_template_rows = ['1fr', '3fr', '1fr']
+
+        all_children = {'header': self.header,
+                        'footer': self.footer,
+                        'left-sidebar': self.left_sidebar,
+                        'right-sidebar': self.right_sidebar,
+                        'center': self.center}
+
+        children = {position : child
+                    for position, child in all_children.items()
+                    if child is not None}
+
+        if not children:
+            return
+
+        for position, child in children.items():
+            child.layout.grid_area = position
+
+        if self.merge:
+
+            if len(children) == 1:
+                position = list(children.keys())[0]
+                grid_template_areas = [[position, position, position],
+                                       [position, position, position],
+                                       [position, position, position]]
+
+            else:
+                if self.center is None:
+                    for row in grid_template_areas:
+                        del row[1]
+                    del grid_template_columns[1]
+
+                if self.left_sidebar is None:
+                    grid_template_areas[1][0] = grid_template_areas[1][1]
+
+                if self.right_sidebar is None:
+                    grid_template_areas[1][-1] = grid_template_areas[1][-2]
+
+                if (self.left_sidebar is None and
+                        self.right_sidebar is None and
+                        self.center is None):
+                    grid_template_areas = [['header'], ['footer']]
+                    grid_template_columns = ['1fr']
+                    grid_template_rows = ['1fr', '1fr']
+
+                if self.header is None:
+                    del grid_template_areas[0]
+                    del grid_template_rows[0]
+
+                if self.footer is None:
+                    del grid_template_areas[-1]
+                    del grid_template_rows[-1]
+
+
+        grid_template_areas_css = "\n".join('"{}"'.format(" ".join(line))
+                                            for line in grid_template_areas)
+
+        self.layout.grid_template_columns = " ".join(grid_template_columns)
+        self.layout.grid_template_rows = " ".join(grid_template_rows)
+        self.layout.grid_template_areas = grid_template_areas_css
+
+        self.children = tuple(children.values())
+
+    @observe("footer", "header", "center", "left_sidebar", "right_sidebar")
```

Comment:
I think you also want to observer "merge".

Reply:
yes, indeed. Good catch. Thanks!